lib/stream.js

```diff
@@ -181,7 +181,7 @@ export const getSpawnedResult = async ({
 		return await Promise.race([
 			Promise.all([
 				undefined,
-				exitPromise,
+				waitForSuccessfulExit(exitPromise),
```

Comment:
Previously, on a non-0 exit code, `Promise.all()` would wait for the streams to complete, then return.
Now, it throws right away, and the `catch` block handles waiting for the streams to complete instead.

The `catch` block ends up doing the exact same thing. 
However, now, that `catch` block can be used to capture any type of error, not just stream errors and process `error` events.